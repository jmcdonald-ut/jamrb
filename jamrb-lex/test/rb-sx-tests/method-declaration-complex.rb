class SomeConst
  # Add method
  def +(other)
  end

  # Subtract method
  def -(other)
  end

  # Multiply method
  def *(other)
  end

  # Power method
  def **(other)
  end

  # Divide other
  def /(other)
  end

  # Modulus other
  def %(other)
  end

  def &(other)
  end

  def ^(other)
  end

  def >>(other)
  end

  def <<(other)
  end

  def >(other)
  end

  def >=(other)
  end

  def <=(other)
  end

  def <(other)
  end

  def !=(other)
  end

  def ==(other)
  end

  def <=>(spaceship_other)
  end

  def !~(other)
  end

  def =~(other)
  end

  def ===(other)
  end

  def [](a)
  end

  def []=(a, b, c)
  end

  def -@
  end

  def +@
  end

  def ~@
  end

  def !@
  end
end
