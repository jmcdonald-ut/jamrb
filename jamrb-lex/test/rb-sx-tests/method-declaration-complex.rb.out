((Token ((1 0) on_kw "class"))
 (Token ((1 5) on_sp " "))
 (Token ((1 6) on_const "SomeConst"))
 (Token ((1 15) on_nl "\n"))
 (Token ((2 0) on_sp "  "))
 (Token ((2 2) on_comment "# Add method\n"))
 (Token ((3 0) on_sp "  "))
 (Token ((3 2) on_kw "def"))
 (Token ((3 5) on_sp " "))
 (Token ((3 6) on_op "+"))
 (Token ((3 7) on_lparen "("))
 (Token ((3 8) on_ident "other"))
 (Token ((3 13) on_rparen ")"))
 (Token ((3 14) on_ignored_nl "\n"))
 (Token ((4 0) on_sp "  "))
 (Token ((4 2) on_kw "end"))
 (Token ((4 5) on_nl "\n"))
 (Token ((5 0) on_ignored_nl "\n"))
 (Token ((6 0) on_sp "  "))
 (Token ((6 2) on_comment "# Subtract method\n"))
 (Token ((7 0) on_sp "  "))
 (Token ((7 2) on_kw "def"))
 (Token ((7 5) on_sp " "))
 (Token ((7 6) on_op "-"))
 (Token ((7 7) on_lparen "("))
 (Token ((7 8) on_ident "other"))
 (Token ((7 13) on_rparen ")"))
 (Token ((7 14) on_ignored_nl "\n"))
 (Token ((8 0) on_sp "  "))
 (Token ((8 2) on_kw "end"))
 (Token ((8 5) on_nl "\n"))
 (Token ((9 0) on_ignored_nl "\n"))
 (Token ((10 0) on_sp "  "))
 (Token ((10 2) on_comment "# Multiply method\n"))
 (Token ((11 0) on_sp "  "))
 (Token ((11 2) on_kw "def"))
 (Token ((11 5) on_sp " "))
 (Token ((11 6) on_op "*"))
 (Token ((11 7) on_lparen "("))
 (Token ((11 8) on_ident "other"))
 (Token ((11 13) on_rparen ")"))
 (Token ((11 14) on_ignored_nl "\n"))
 (Token ((12 0) on_sp "  "))
 (Token ((12 2) on_kw "end"))
 (Token ((12 5) on_nl "\n"))
 (Token ((13 0) on_ignored_nl "\n"))
 (Token ((14 0) on_sp "  "))
 (Token ((14 2) on_comment "# Power method\n"))
 (Token ((15 0) on_sp "  "))
 (Token ((15 2) on_kw "def"))
 (Token ((15 5) on_sp " "))
 (Token ((15 6) on_op "**"))
 (Token ((15 8) on_lparen "("))
 (Token ((15 9) on_ident "other"))
 (Token ((15 14) on_rparen ")"))
 (Token ((15 15) on_ignored_nl "\n"))
 (Token ((16 0) on_sp "  "))
 (Token ((16 2) on_kw "end"))
 (Token ((16 5) on_nl "\n"))
 (Token ((17 0) on_ignored_nl "\n"))
 (Token ((18 0) on_sp "  "))
 (Token ((18 2) on_comment "# Divide other\n"))
 (Token ((19 0) on_sp "  "))
 (Token ((19 2) on_kw "def"))
 (Token ((19 5) on_sp " "))
 (Token ((19 6) on_op "/"))
 (Token ((19 7) on_lparen "("))
 (Token ((19 8) on_ident "other"))
 (Token ((19 13) on_rparen ")"))
 (Token ((19 14) on_ignored_nl "\n"))
 (Token ((20 0) on_sp "  "))
 (Token ((20 2) on_kw "end"))
 (Token ((20 5) on_nl "\n"))
 (Token ((21 0) on_ignored_nl "\n"))
 (Token ((22 0) on_sp "  "))
 (Token ((22 2) on_comment "# Modulus other\n"))
 (Token ((23 0) on_sp "  "))
 (Token ((23 2) on_kw "def"))
 (Token ((23 5) on_sp " "))
 (Token ((23 6) on_op "%"))
 (Token ((23 7) on_lparen "("))
 (Token ((23 8) on_ident "other"))
 (Token ((23 13) on_rparen ")"))
 (Token ((23 14) on_ignored_nl "\n"))
 (Token ((24 0) on_sp "  "))
 (Token ((24 2) on_kw "end"))
 (Token ((24 5) on_nl "\n"))
 (Token ((25 0) on_ignored_nl "\n"))
 (Token ((26 0) on_sp "  "))
 (Token ((26 2) on_kw "def"))
 (Token ((26 5) on_sp " "))
 (Token ((26 6) on_op "&"))
 (Token ((26 7) on_lparen "("))
 (Token ((26 8) on_ident "other"))
 (Token ((26 13) on_rparen ")"))
 (Token ((26 14) on_ignored_nl "\n"))
 (Token ((27 0) on_sp "  "))
 (Token ((27 2) on_kw "end"))
 (Token ((27 5) on_nl "\n"))
 (Token ((28 0) on_ignored_nl "\n"))
 (Token ((29 0) on_sp "  "))
 (Token ((29 2) on_kw "def"))
 (Token ((29 5) on_sp " "))
 (Token ((29 6) on_op "^"))
 (Token ((29 7) on_lparen "("))
 (Token ((29 8) on_ident "other"))
 (Token ((29 13) on_rparen ")"))
 (Token ((29 14) on_ignored_nl "\n"))
 (Token ((30 0) on_sp "  "))
 (Token ((30 2) on_kw "end"))
 (Token ((30 5) on_nl "\n"))
 (Token ((31 0) on_ignored_nl "\n"))
 (Token ((32 0) on_sp "  "))
 (Token ((32 2) on_kw "def"))
 (Token ((32 5) on_sp " "))
 (Token ((32 6) on_op ">>"))
 (Token ((32 8) on_lparen "("))
 (Token ((32 9) on_ident "other"))
 (Token ((32 14) on_rparen ")"))
 (Token ((32 15) on_ignored_nl "\n"))
 (Token ((33 0) on_sp "  "))
 (Token ((33 2) on_kw "end"))
 (Token ((33 5) on_nl "\n"))
 (Token ((34 0) on_ignored_nl "\n"))
 (Token ((35 0) on_sp "  "))
 (Token ((35 2) on_kw "def"))
 (Token ((35 5) on_sp " "))
 (Token ((35 6) on_op "<<"))
 (Token ((35 8) on_lparen "("))
 (Token ((35 9) on_ident "other"))
 (Token ((35 14) on_rparen ")"))
 (Token ((35 15) on_ignored_nl "\n"))
 (Token ((36 0) on_sp "  "))
 (Token ((36 2) on_kw "end"))
 (Token ((36 5) on_nl "\n"))
 (Token ((37 0) on_ignored_nl "\n"))
 (Token ((38 0) on_sp "  "))
 (Token ((38 2) on_kw "def"))
 (Token ((38 5) on_sp " "))
 (Token ((38 6) on_op ">"))
 (Token ((38 7) on_lparen "("))
 (Token ((38 8) on_ident "other"))
 (Token ((38 13) on_rparen ")"))
 (Token ((38 14) on_ignored_nl "\n"))
 (Token ((39 0) on_sp "  "))
 (Token ((39 2) on_kw "end"))
 (Token ((39 5) on_nl "\n"))
 (Token ((40 0) on_ignored_nl "\n"))
 (Token ((41 0) on_sp "  "))
 (Token ((41 2) on_kw "def"))
 (Token ((41 5) on_sp " "))
 (Token ((41 6) on_op ">="))
 (Token ((41 8) on_lparen "("))
 (Token ((41 9) on_ident "other"))
 (Token ((41 14) on_rparen ")"))
 (Token ((41 15) on_ignored_nl "\n"))
 (Token ((42 0) on_sp "  "))
 (Token ((42 2) on_kw "end"))
 (Token ((42 5) on_nl "\n"))
 (Token ((43 0) on_ignored_nl "\n"))
 (Token ((44 0) on_sp "  "))
 (Token ((44 2) on_kw "def"))
 (Token ((44 5) on_sp " "))
 (Token ((44 6) on_op "<="))
 (Token ((44 8) on_lparen "("))
 (Token ((44 9) on_ident "other"))
 (Token ((44 14) on_rparen ")"))
 (Token ((44 15) on_ignored_nl "\n"))
 (Token ((45 0) on_sp "  "))
 (Token ((45 2) on_kw "end"))
 (Token ((45 5) on_nl "\n"))
 (Token ((46 0) on_ignored_nl "\n"))
 (Token ((47 0) on_sp "  "))
 (Token ((47 2) on_kw "def"))
 (Token ((47 5) on_sp " "))
 (Token ((47 6) on_op "<"))
 (Token ((47 7) on_lparen "("))
 (Token ((47 8) on_ident "other"))
 (Token ((47 13) on_rparen ")"))
 (Token ((47 14) on_ignored_nl "\n"))
 (Token ((48 0) on_sp "  "))
 (Token ((48 2) on_kw "end"))
 (Token ((48 5) on_nl "\n"))
 (Token ((49 0) on_ignored_nl "\n"))
 (Token ((50 0) on_sp "  "))
 (Token ((50 2) on_kw "def"))
 (Token ((50 5) on_sp " "))
 (Token ((50 6) on_op "!="))
 (Token ((50 8) on_lparen "("))
 (Token ((50 9) on_ident "other"))
 (Token ((50 14) on_rparen ")"))
 (Token ((50 15) on_ignored_nl "\n"))
 (Token ((51 0) on_sp "  "))
 (Token ((51 2) on_kw "end"))
 (Token ((51 5) on_nl "\n"))
 (Token ((52 0) on_ignored_nl "\n"))
 (Token ((53 0) on_sp "  "))
 (Token ((53 2) on_kw "def"))
 (Token ((53 5) on_sp " "))
 (Token ((53 6) on_op "=="))
 (Token ((53 8) on_lparen "("))
 (Token ((53 9) on_ident "other"))
 (Token ((53 14) on_rparen ")"))
 (Token ((53 15) on_ignored_nl "\n"))
 (Token ((54 0) on_sp "  "))
 (Token ((54 2) on_kw "end"))
 (Token ((54 5) on_nl "\n"))
 (Token ((55 0) on_ignored_nl "\n"))
 (Token ((56 0) on_sp "  "))
 (Token ((56 2) on_kw "def"))
 (Token ((56 5) on_sp " "))
 (Token ((56 6) on_op "<=>"))
 (Token ((56 9) on_lparen "("))
 (Token ((56 10) on_ident "spaceship_other"))
 (Token ((56 25) on_rparen ")"))
 (Token ((56 26) on_ignored_nl "\n"))
 (Token ((57 0) on_sp "  "))
 (Token ((57 2) on_kw "end"))
 (Token ((57 5) on_nl "\n"))
 (Token ((58 0) on_ignored_nl "\n"))
 (Token ((59 0) on_sp "  "))
 (Token ((59 2) on_kw "def"))
 (Token ((59 5) on_sp " "))
 (Token ((59 6) on_op "!~"))
 (Token ((59 8) on_lparen "("))
 (Token ((59 9) on_ident "other"))
 (Token ((59 14) on_rparen ")"))
 (Token ((59 15) on_ignored_nl "\n"))
 (Token ((60 0) on_sp "  "))
 (Token ((60 2) on_kw "end"))
 (Token ((60 5) on_nl "\n"))
 (Token ((61 0) on_ignored_nl "\n"))
 (Token ((62 0) on_sp "  "))
 (Token ((62 2) on_kw "def"))
 (Token ((62 5) on_sp " "))
 (Token ((62 6) on_op "=~"))
 (Token ((62 8) on_lparen "("))
 (Token ((62 9) on_ident "other"))
 (Token ((62 14) on_rparen ")"))
 (Token ((62 15) on_ignored_nl "\n"))
 (Token ((63 0) on_sp "  "))
 (Token ((63 2) on_kw "end"))
 (Token ((63 5) on_nl "\n"))
 (Token ((64 0) on_ignored_nl "\n"))
 (Token ((65 0) on_sp "  "))
 (Token ((65 2) on_kw "def"))
 (Token ((65 5) on_sp " "))
 (Token ((65 6) on_op "==="))
 (Token ((65 9) on_lparen "("))
 (Token ((65 10) on_ident "other"))
 (Token ((65 15) on_rparen ")"))
 (Token ((65 16) on_ignored_nl "\n"))
 (Token ((66 0) on_sp "  "))
 (Token ((66 2) on_kw "end"))
 (Token ((66 5) on_nl "\n"))
 (Token ((67 0) on_ignored_nl "\n"))
 (Token ((68 0) on_sp "  "))
 (Token ((68 2) on_kw "def"))
 (Token ((68 5) on_sp " "))
 (Token ((68 6) on_op "[]"))
 (Token ((68 8) on_lparen "("))
 (Token ((68 9) on_ident "a"))
 (Token ((68 10) on_rparen ")"))
 (Token ((68 11) on_ignored_nl "\n"))
 (Token ((69 0) on_sp "  "))
 (Token ((69 2) on_kw "end"))
 (Token ((69 5) on_nl "\n"))
 (Token ((70 0) on_ignored_nl "\n"))
 (Token ((71 0) on_sp "  "))
 (Token ((71 2) on_kw "def"))
 (Token ((71 5) on_sp " "))
 (Token ((71 6) on_op "[]="))
 (Token ((71 9) on_lparen "("))
 (Token ((71 10) on_ident "a"))
 (Token ((71 11) on_comma ","))
 (Token ((71 12) on_sp " "))
 (Token ((71 13) on_ident "b"))
 (Token ((71 14) on_comma ","))
 (Token ((71 15) on_sp " "))
 (Token ((71 16) on_ident "c"))
 (Token ((71 17) on_rparen ")"))
 (Token ((71 18) on_ignored_nl "\n"))
 (Token ((72 0) on_sp "  "))
 (Token ((72 2) on_kw "end"))
 (Token ((72 5) on_nl "\n"))
 (Token ((73 0) on_ignored_nl "\n"))
 (Token ((74 0) on_sp "  "))
 (Token ((74 2) on_kw "def"))
 (Token ((74 5) on_sp " "))
 (Token ((74 6) on_op "-@"))
 (Token ((74 8) on_nl "\n"))
 (Token ((75 0) on_sp "  "))
 (Token ((75 2) on_kw "end"))
 (Token ((75 5) on_nl "\n"))
 (Token ((76 0) on_ignored_nl "\n"))
 (Token ((77 0) on_sp "  "))
 (Token ((77 2) on_kw "def"))
 (Token ((77 5) on_sp " "))
 (Token ((77 6) on_op "+@"))
 (Token ((77 8) on_nl "\n"))
 (Token ((78 0) on_sp "  "))
 (Token ((78 2) on_kw "end"))
 (Token ((78 5) on_nl "\n"))
 (Token ((79 0) on_ignored_nl "\n"))
 (Token ((80 0) on_sp "  "))
 (Token ((80 2) on_kw "def"))
 (Token ((80 5) on_sp " "))
 (Token ((80 6) on_op "~@"))
 (Token ((80 8) on_nl "\n"))
 (Token ((81 0) on_sp "  "))
 (Token ((81 2) on_kw "end"))
 (Token ((81 5) on_nl "\n"))
 (Token ((82 0) on_ignored_nl "\n"))
 (Token ((83 0) on_sp "  "))
 (Token ((83 2) on_kw "def"))
 (Token ((83 5) on_sp " "))
 (Token ((83 6) on_op "!@"))
 (Token ((83 8) on_nl "\n"))
 (Token ((84 0) on_sp "  "))
 (Token ((84 2) on_kw "end"))
 (Token ((84 5) on_nl "\n"))
 (Token ((85 0) on_kw "end"))
 (Token ((85 3) on_nl "\n")))
